### ISSUE TITLE
phase out manual use of qubes-dom0-update by user / replace it by salt

### ISSUE THREAD

**Issue by adrelanos** [QubesOS/qubes-issues#3447]
Talked to Marek at 34c3. We concluded the following.

* Installing Whonix manually using `qubes-dom0-update` got too difficult since too many manual modifications of dom0 would be required.
* Most users shall no longer install Whonix using `qubes-dom0-update`.
  * Most users shall install Whonix using Qubes `salt`.
* Most users shall no longer update using `qubes-dom0-update`.
  * Should use `sudo qubesctl --templates pkg.upgrade` instead.
* We need `salt` commands for installation of arch linux and others
* Modify `qubes-dom0-update` to tell the user it is recommended to use `salt` instead.

Basically, let's phase out manual use of `qubes-dom0-update` by users.

Technical background:

* Marek said we cannot make `qubes-dom0-update` run `salt`, because `salt` is running `qubes-dom0-update` and it must not be circular. Would be tricky because not clear which salt `salt` formula to enable.
* This is a description what Qubes `salt` does (confirmed by Marek): https://www.whonix.org/wiki/Dev/Qubes#salt

**Comment by tasket** [comment]
Then `salt` will still use `qubes-dom0-update` in R4.1...?

**Comment by adrelanos** [comment]
Yes, `salt` will continue to call `qubes-dom0-update`.

**Comment by adrelanos** [comment]
//cc @awokd

**Comment by awokd** [comment]
Thanks for the heads up @adrelanos , hadn't seen this. I linked #1983 from the R4.0 doc tracking ticket.

**Comment by andrewdavidwong** [comment]
Per https://github.com/QubesOS/qubes-issues/issues/3595#issuecomment-367868742, this issue should be understood more generally to include things like the creation of `whonix-ws`-based VMs, not as limited solely to `qubes-dom0-update`.

**Comment by ben-grande** [comment]
> Basically, let's phase out manual use of `qubes-dom0-update` by users.

For command-line:

- qubes-dom0-update
- salt
- ansible
- qvm-template
- qubes-vm-update

For GUI:

- qvm-template-gui
- qubes-update-gui

But of these tools, few allow package installation on dom0 (excluding installing templates as packages):

- qubes-dom0-update
- salt
- ansible

And neither Ansible nor Salt are replacing `qubes-dom0-update`. Both call it under the hood. When writing the docs, it is easier to provide `qubes-dom0-update` to the command-line than telling the user to copy the salt/ansible yaml to dom0 and run this salt/ansible command.
